okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java

```diff
@@ -178,7 +178,8 @@ private QosIoExceptionAwareOkHttpClient(OkHttpClient delegate, Supplier<QosIoExc
 
         @Override
         public Call newCall(Request request) {
-            return new QosIoExceptionAwareCall(getDelegate().newCall(request), handlerFactory.get());
+            return new InterruptibleCall(
+                    new QosIoExceptionAwareCall(getDelegate().newCall(request), handlerFactory.get()));
```

Comment:
do we need to layer these the other way so that we exit from the interrupted thread earlier?

Reply:
I think since we call the async codepath on the `QosIoExceptionAwareCall` from inside `InterruptibleCall`, then catch the `InterruptedException` while waiting on a future for that async work to finish, it will handle the interruption just as soon as if the layering was the other way. I tested layering them the other way and it does still pass the test, though.